internal/generateFlatbuffer.ts

```diff
@@ -0,0 +1,137 @@
+import { FoxgloveMessageField, FoxglovePrimitive, FoxgloveSchema } from "./types";
+
+// Flatbuffer only supports nested vectors via table
+export const BYTE_VECTOR_FB = `table ByteVectorForNesting {
+  data:[byte];
+}
+
+root_type ByteVectorForNesting;
+`;
+
+// Same as protobuf wellknown types
+export const TIME_FB = `struct Time {
+  sec:long;
+  nsec:int;
+}
+`;
+
+export const DURATION_FB = `struct Duration {
+  sec:long;
+  nsec:int;
+}
+`;
+
+// fields that would benefit from having a default of 1
+const defaultOneNumberFields = new Set(["x", "y", "z", "r", "g", "b", "a", "w"]);
+function isDefaultOneField(field: FoxgloveMessageField): boolean {
+  return (
+    field.type.type === "primitive" &&
+    field.type.name === "float64" &&
+    defaultOneNumberFields.has(field.name)
+  );
+}
+
+function primitiveToFlatbuffer(type: Exclude<FoxglovePrimitive, "time" | "duration">) {
+  switch (type) {
+    case "uint32":
+      return "uint";
+    case "bytes":
+      return "[byte]";
+    case "string":
+      return "string";
+    case "boolean":
+      return "bool";
+    case "float64":
+      return "double";
+  }
+}
+
+export function generateFlatbuffer(schema: FoxgloveSchema): string {
+  let definition;
+  const imports = new Set<string>();
+  switch (schema.type) {
+    case "enum": {
+      const fields = schema.values.map(({ name, value, description }) => {
+        if (description != undefined) {
+          return `/// ${description}\n  ${name} = ${value},`;
+        } else {
+          return `${name} = ${value},`;
+        }
+      });
+
+      // `///` comments required to show up in compiled flatbuffer schemas
+      definition = `/// ${schema.description}\nenum ${schema.name} : byte {\n  ${fields.join(
```

Comment:
Should we use ubyte for enums? Might want to add validation that the values all fit within the range of the type we choose. I assume the compiler would validate this, so if we can get the compiler to be tested in CI, then that would be sufficient.